### PR TITLE
Adds the M-22 bayonet (knife), and redoes shoe storage while at it

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Adds a knife that can become a bayonet. And re-does shoe storage while at it.
Obj sprites are from Kuro! Big thanks to him.

## Why It's Good For The Game
Wealism.
Also because it sounded neat in my head.

## Changelog
:cl:
add: Added a knife which can be a bayonet.
tweak: Changed how marine boots storage works to pockets.
balance: Added a knife which can work as a bayonet.
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -4. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by Pariah919** [comment]
>You currently have a negative Fix/Feature pull request delta of -4. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

The heck do you want bot smh

**Comment by terranaut1** [comment]
ignoring that this is completely unnecessary as bayonets already fucking exist and are free and knives are free and go in your boot already so the most optimal solution for people who want to use a bayonet is to get the bayonet and keep their knife in the boot, the move you should do here is just making the existing combat knife attachable instead of bloating things with extra shit that aren't needed or confusing players unnecessarily.
i (and a bunch of others) already said this on discord, this is just for the maintainers who will hopefully not merge it as he initially proposed it

**Comment by Pariah919** [comment]
I don't personally see this as bloat. But if they consider it to be bloat then I suppose it is.

File: code/modules/clothing/shoes/marine_shoes.dm [hunk 1]
```diff
@@ -12,43 +12,54 @@
...
-	if(knife)
-		QDEL_NULL(knife)
-	return ..()
+/obj/item/storage/internal/shoes/boot_knife
+    max_storage_space = 3
+    storage_slots = 1
+    can_hold = list(
+        /obj/item/weapon/combat_knife,
+        /obj/item/attachable/bayonetknife,
+		/obj/item/weapon/throwing_knife
```
Comment (psykzz) [inline code/modules/clothing/shoes/marine_shoes.dm:23]: indentation

**Comment by psykzz** [comment]
From what i can see this replaces the combat knife with the T22 and removes the bayonet since the t22 can be attached directly?

My only feedback would be to somehow rework attachments to convert their types when attached so we didn't have to have the knife on the `/obj/item/attachable` path, but thats a pretty significate change.

**Comment by Pariah919** [comment]
Seems to conflict with the sg replacement. So I suppose this needs that merged before this can be tmed.

**Comment by jroinc** [comment]
You know you can currently convert knives into bayonets with a cable coil, right?

This is bloat. If you want to make it, remove bayonets and let the survival knife just attach regardless.

**Comment by Pariah919** [comment]
Got new, better obj sprites from Kuro. Otherwise, this should be done.

**Comment by GravitronBlue** [comment]
👎 Due to clicking boots opening up a one slot storage inventory instead of just handing me my knife. 
(Alt-click also does nothing even though on TG it draws items in storage)

File: code/modules/clothing/shoes/marine_shoes.dm [hunk 2]
```diff
@@ -12,40 +12,51 @@
...
-//ATTACK HAND IGNORING PARENT RETURN VALUE
 /obj/item/clothing/shoes/marine/attack_hand(mob/living/user)
-	if(knife && src.loc == user && !user.incapacitated()) //Only allow someone to take out the knife if it's being worn or held. So you can pick them up off the floor
-		if(user.put_in_active_hand(knife))
-			to_chat(user, "<span class='notice'>You slide [knife] out of [src].</span>")
-			playsound(user, 'sound/weapons/guns/interact/shotgun_shell_insert.ogg', 15, 1)
-			knife = 0
-			update_icon()
-	else
+	if(!pockets)
```
Comment (Rohesie) [inline code/modules/clothing/shoes/marine_shoes.dm:31]: Should this ever lack pockets?
Also, shouldn't the click only transfer when held? You don't want to click on the container when trying to grab the boots from the ground.

File: code/modules/projectiles/gun_attachables.dm [hunk 3]
```diff
@@ -341,6 +341,28 @@ Defined in conflicts.dm of the #defines folder.
...
+	attach_icon = "bayonetknife_a"
+	force = 25
+	throwforce = 20
+	throw_speed = 3
+	throw_range = 6
+	attack_speed = 7
+	attach_delay = 10 //Bayonets attach/detach quickly.
+	detach_delay = 10
+	attack_verb = list("slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
+	melee_mod = 25
```
Comment (Rohesie) [inline code/modules/projectiles/gun_attachables.dm:358]:
```suggestion
	melee_mod = 20
```
Suggestion.

File: code/modules/clothing/shoes/marine_shoes.dm (hunk 2 above)
Comment (Pariah919) [reply code/modules/clothing/shoes/marine_shoes.dm]: I removed the redundancy check.
It already requires you to click and hold to you if you need to get into the internal inventory that way if that's what you mean.

File: code/modules/projectiles/gun_attachables.dm (hunk 3 above)
Comment (Pariah919) [reply code/modules/projectiles/gun_attachables.dm:358]: Eh. I personally like it being basically a bayonet that can be put in your pocket or use as a knife. . As I replaced bayonets inside the combat vendors with this.

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-7). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**